src/Repository/ConnectedGithubClient.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace OAT\DependencyResolver\Repository;
+
+use Github\Exception\ErrorException;
+use Github\Exception\RuntimeException;
+use OAT\DependencyResolver\Repository\Entity\Repository;
+use OAT\DependencyResolver\Repository\Exception\BranchNotFoundException;
+use OAT\DependencyResolver\Repository\Exception\EmptyRepositoryException;
+use OAT\DependencyResolver\Repository\Exception\FileNotFoundException;
+
+class ConnectedGithubClient
+{
+    /** @var GithubClientProxy */
+    protected $client;
+
+    /** @var string */
+    protected $organization = '';
+
+    /** @var string */
+    protected $token = '';
+
+    /**
+     * Is the client authenticated?
+     *
+     * @var bool
+     */
+    protected $authenticated = false;
+
+    /**
+     * Stores the numbers of public and private repositories.
+     *
+     * @var array
+     */
+    protected $organisationProperties = [];
+
+    public function __construct(GithubClientProxy $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Stores token for github client.
+     */
+    public function setToken(string $token)
+    {
+        $this->token = $token;
+    }
+
+    public function getOrganizationProperties(string $owner): array
+    {
+        $this->authenticateAndCheck($owner);
+
+        return $this->organisationProperties;
+    }
+
+    /**
+     * Retrieves organization's repositories.
+     *
+     * @param string $owner
+     * @param int    $perPage Number of repositories to load per client call.
+     *
+     * @return array
+     */
+    public function getRepositoryList(string $owner, int $perPage = 100): array
```

Comment:
I think owner should come from configuration (environment variable)

Reply:
It is. It's injected from GitHubRepositoryReader < RepositoryMapUpdater < UpdateRepositoryMapCommand < service.yaml